### ISSUE TITLE
Disabling a Metric until first non-None output from inference()

### ISSUE THREAD

**Issue by g-karthik** [pytorch/ignite#1310]
## ❓ Questions/Help/Support

So I have an `inference()` method as such:

```
def inference(engine, batch):
     # do forward() with model
     
     return (lm_logits, mc_logits), (lm_labels, mc_labels)

validator = Engine(inference)
```

Now I have a metric like this:

```
metric = Accuracy(output_transform=lambda x: (x[0][1], x[1][1]))
metric.attach(validator, "accuracy")
```

Now in my use-case, there can be situations where the `mc_logits` and `mc_labels` returned by `inference()` are `None`, in which case the Accuracy metric cannot be computed. For example, if I am using a model that does not perform multiple-choice classification, and only performs language modeling.

So what I want to ensure is that the `Accuracy` metric is "disabled" until the first `batch` arrives such that `mc_logits` and `mc_labels` are non-None. Therefore, in case of models that only perform language modeling, this will always be None and so the `Accuracy` metric would never get "enabled".

Is there a simple way to accomplish this?

@vfdev-5

**Comment by vfdev-5** [comment]
@g-karthik maybe this is related to #1065

However, we can skip metric update by using metric's "usage" option like that 
```python
import torch

from ignite.metrics import Accuracy
from ignite.engine import Engine

from ignite.metrics.metric import BatchFiltered

mc_logits_list = torch.ones(20, 1, 10)
mc_labels_list = [0, None, 2, 3, None, None, 6, 7, 8, None, 0, 1, 2, None, None, None, 6, 7, None, 9]
# mc_labels_list = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 0, 1, 2, 3, 4, 5, 6, 7, 8, 9]

def inference_step(e, b):
    i = e.state.iteration - 1
    mc_logits = mc_logits_list[i, :, :]
    mc_labels = mc_labels_list[i]
    if mc_labels is not None:
        mc_labels = torch.tensor([mc_labels])
    return mc_logits, mc_labels

def filterout_nones(e, ev):
    if e.state.output[1] is None:
        print("Skip this output: ", e.state.output)
        return False
    return True

inferencer = Engine(inference_step)
acc = Accuracy()
acc.attach(inferencer, "acc", usage=BatchFiltered(event_filter=filterout_nones))

state = inferencer.run(range(20))
print(state.metrics["acc"])
```

**Comment by g-karthik** [comment]
@vfdev-5 

Thanks! I have an another `Metric` to compute the all-reduced accuracy, by leveraging the `Accuracy()`. Should I use `BatchFiltered` for the other `Metric` also when attaching it?

I think not? It's just for the single-process `Accuracy()` attach?

**Comment by vfdev-5** [comment]
If no need to skip metric updates for another accuracy, attach without
specific "usage" option.

**Comment by vfdev-5** [comment]
> EDIT: Also, I have ignite 0.4.1, but I do not see the event_filter argument to the BatchFiltered class. Is that a new arg yet to be released?

`event_filter` is an argument of `Events.*` and `BatchFiltered` accepts args/kwargs to configure events filtering: 
https://pytorch.org/ignite/v0.4.1/metrics.html#ignite.metrics.BatchFiltered
